### PR TITLE
[better-engineering] use AT_DISPATCH_V2 macro

### PR DESCRIPTION
**Pull request by khushi-411**
cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @chauhang @d4l3k @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire

### CODE REVIEW

File: aten/src/ATen/native/cuda/UnaryGeometricSinKernel.cu [hunk 1]
```diff
@@ -22,33 +22,43 @@ void sin_kernel_cuda(TensorIteratorBase& iter) {
...
+        "sin_name",
+        AT_WRAP([&] {
           jitted_gpu_kernel<
               /*name=*/sin_name,
               /*return_dtype=*/scalar_t,
               /*common_dtype=*/scalar_t,
               /*arity=*/1>(iter, sin_string);
-        });
+        }),
+        kComplexHalf,
```
Comment (ezyang) [inline aten/src/ATen/native/cuda/UnaryGeometricSinKernel.cu:35]: you forgot AT_EXPAND(AT_COMPLEX_TYPES) here

**Review by ezyang — CHANGES_REQUESTED** [review]
looks good, just some errors

File: aten/src/ATen/native/cuda/UnaryGeometricSinKernel.cu (hunk 1 above)
Comment (khushi-411) [reply aten/src/ATen/native/cuda/UnaryGeometricSinKernel.cu:36]: My bad 😅

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
something bad happened. Open a new PR please

**Pull request closed without merging**